### ISSUE TITLE
[P1] W&B Artifacts provenance — full lineage for reproducibility

### ISSUE THREAD

**Issue by fritzprix** [fritzprix/systemone-lite#11]
# Issue: W&B Artifacts provenance — full lineage for reproducibility

## Problem

Run `spatial-v2-s1-cloze` has **no W&B Artifacts** for:
- Dataset used
- Base model used
- Resume checkpoint used
- Training code
- Output checkpoints

Everything is referenced by local paths, making the run **unreproducible** and **unauditable**.

## Postmortem Evidence

From the postmortem:
- Data is only available as local paths
- Start model is a local checkpoint path
- No W&B input artifact used
- Training code (`chess_finetune.py`) was not captured
- Single seed (7) — no seed diversity

This means:
1. Cannot reproduce this exact experiment
2. Cannot verify what code was actually run
3. Cannot audit which data was used
4. Cannot resume from W&B (no artifact lineage)

### 1. W&B Artifacts pipeline

```python
import wandb

# Before training starts:
run = wandb.init(...)

# Upload dataset as artifact
dataset_artifact = wandb.Artifact("dataset-phase2-v2", type="dataset")
dataset_artifact.add_dir("data/")
run.log_artifact(dataset_artifact)

# Upload base model
base_model_artifact = wandb.Artifact("base-qwen2.5-0.5b", type="model")
base_model_artifact.add_file("checkpoints/qwen2.5-0.5b/")
run.log_artifact(base_model_artifact)

# Upload training code
code_artifact = wandb.Artifact("training-code", type="code")
code_artifact.add_file("scripts/chess_finetune.py")
code_artifact.add_file("src/systemone_lite/")
run.log_artifact(code_artifact)

# On checkpoint save:
ckpt_artifact = wandb.Artifact(f"checkpoint-step-{step}", type="checkpoint")
ckpt_artifact.add_file(f"checkpoints/step_{step}.pt")
run.log_artifact(ckpt_artifact)
```

### 2. Artifact lineage for resume

```python
# On resume:
run = wandb.init(...)
artifact = run.use_artifact("checkpoint-step-20000:latest", type="checkpoint")
artifact_dir = artifact.download()
resume_from = os.path.join(artifact_dir, "step_20000.pt")
```

### 3. Required artifact types

| Type | When | Contents |
|---|---|---|
| `dataset` | Start of training | JSONL files, config.yaml |
| `model` | Start of training | Base checkpoint or HF model cache |
| `code` | Start of training | `chess_finetune.py`, `src/`, `config.yaml` |
| `checkpoint` | Every save_every | `step_NNNNN.pt` |
| `output` | End of training | Final checkpoint, logs, metrics summary |

### 4. Code capture requirement

**Always** capture the exact training script:
```bash
wandb code scripts/chess_finetune.py
# or via artifact
```

## Acceptance Criteria

- [ ] Dataset uploaded as W&B artifact before training starts
- [ ] Base model uploaded as W&B artifact before training starts
- [ ] Training code uploaded as W&B artifact before training starts
- [ ] Checkpoints uploaded as W&B artifacts on every save
- [ ] Resume uses `run.use_artifact()` instead of local path
- [ ] Final output checkpoint uploaded as W&B artifact
- [ ] `scripts/chess_finetune.py` is always the code artifact (not a copy)
